### PR TITLE
Adding support of negative values for line chart 

### PR DESCRIPTION
**Pull request by cherurg**
## Description

Following the advice in #338, I've made the modification in `line.js` file. In addition, I've introduced `horizontal-grid-line--highlighted` CSS class that highlights the 0-axis in case there are negative values in the chart. Also I've changed how styles are imported in sandbox (see sandbox/index.html and sandbox/sandbox.js changes). Personally I don't care too much about them but I found it's easier to use sandbox if styles are re-compiled dynamically. Let me know if you want to exclude this change completely or if you want me to create another PR for this.  

## Motivation and Context

Fixed this issue: https://github.com/eventbrite/britecharts/issues/338

## How Has This Been Tested?

I've created three tests.
1. Check that the first Y-tick is negative if there are negative values.
2. Check that the 0-axis has `horizontal-grid-line--highlighted` class if there are negative values.
3. Check that the 0-axis has NO `horizontal-grid-line--highlighted` class if there is NO negative values.

## Screenshots:

Screenshot 2020-02-21 at 14 21 23

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] Latest master code has been merged into this branch
- [ ] No commented out code (if required, place // TODO above with explanation)
- [ ] No linting issues
- [ ] Build is successful
- [ ] Code follows the [API Guidelines](http://eventbrite.github.io/britecharts/topics-index.html#toc5__anchor)
- [ ] Updated the documentation
- [ ] Added tests to cover changes
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by DalerAsrorov** [comment]
This is beautiful, my friend.

File: sandbox/index.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <html>
     <head>
         <title>Britecharts Sandbox</title>
-        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/britecharts/dist/css/britecharts.min.css" type="text/css" />
+        <!-- <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/britecharts/dist/css/britecharts.min.css" type="text/css" /> -->
```
Comment (DalerAsrorov) [inline sandbox/index.html]: Why is this commented out?

Comment (cherurg) [reply sandbox/index.html]: Hi! I’m glad you like the PR in general.
Regarding this particular line, I have a comment in the main post. 

> Also I've changed how styles are imported in sandbox (see sandbox/index.html and sandbox/sandbox.js changes). Personally I don't care too much about them but I found it's easier to use sandbox if styles are re-compiled dynamically. Let me know if you want to exclude this change completely or if you want me to create another PR for this.

Comment (DalerAsrorov) [reply sandbox/index.html]: Cool, you can remove this. I like the import approach you did in the `sandbox.js`.

File: sandbox/sandbox.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const constants = require('./constants/constants');
 const defaultData = require('./constants/data');
 const storage = require('./helpers/localStorageHelpers');
 const domHelpers = require('./helpers/domHelpers');
+require('../src/styles/britecharts.scss');
```
Comment (DalerAsrorov) [inline sandbox/sandbox.js]: Can add a line after `domHelpers` constant?

**Comment by cherurg** [comment]
@DalerAsrorov like that?

**Comment by DalerAsrorov** [comment]
looks good @cherurg  let @Golodhros do final review before we do merge.

File: src/charts/line.js [hunk 3]
```diff
@@ -750,13 +750,17 @@ define(function(require){
...
+            let shouldHighlight0Axis = minY < 0;
+            
             if (grid === 'horizontal' || grid === 'full') {
                 horizontalGridLines = svg.select('.grid-lines-group')
                     .selectAll('line.horizontal-grid-line')
                     .data(yScale.ticks(yTicks))
                     .enter()
                       .append('line')
-                        .attr('class', 'horizontal-grid-line')
+                        .classed('horizontal-grid-line', true)
```
Comment (Golodhros) [inline src/charts/line.js]: This is a bit of a nit, but I favor using attr('class', ...) when there is no logic involved on adding the class.

File: src/charts/line.js [hunk 4]
```diff
@@ -750,13 +750,17 @@ define(function(require){
                 .selectAll('line')
                 .remove();
 
+            let minY = d3Array.min(dataByTopic, ({dates}) => d3Array.min(dates, getValue));
+            let shouldHighlight0Axis = minY < 0;
```
Comment (Golodhros) [inline src/charts/line.js]: Nit: I would rename it to shouldHighlightXAxis

File: src/styles/common/_grid.scss [hunk 5]
```diff
@@ -7,6 +7,11 @@
     stroke-dasharray: 4, 4;
 }
 
+.horizontal-grid-line--highlighted {
+    stroke: #ccc;
```
Comment (Golodhros) [inline src/styles/common/_grid.scss]: Please, use one of the grey color variables we declare in https://github.com/Golodhros/britecharts/blob/master/src/styles/helpers/_variables.scss#L14 instead

**Comment by cherurg** [comment]
@golodhros
Thank you for the review! I’ll add the changes you requested a bit later and update the PR

**Comment by Golodhros** [comment]
Thanks a lot for this!!

I will trigger a release soon and also add it to our Version 3 RC.

Welcome to the Britecharts' contributors family!

File: src/styles/common/_grid.scss [hunk 6]
```diff
@@ -7,6 +7,11 @@
     stroke-dasharray: 4, 4;
 }
 
+.horizontal-grid-line--highlighted {
+    stroke: $grey-700;
```
Comment (DalerAsrorov) [inline src/styles/common/_grid.scss]: Nice!

**Pull request merged**